pkg/proxy/register.go

```diff
@@ -54,7 +55,17 @@ func (p *Register) Add(route *Route) error {
 		p.doRegister(matcher.Extract(definition.ListenPath), handler.ServeHTTP, definition.Methods, route.Inbound)
 	}
 
-	p.doRegister(definition.ListenPath, handler.ServeHTTP, definition.Methods, route.Inbound)
+	handle := handler.ServeHTTP
+	if p.params.Prometheus.Enabled {
```

Comment:
I would put this in a plugin that can based on the configurations, enable or disable Prometheus.